### ISSUE TITLE
Steam gobbling shut down request ?

### ISSUE THREAD

**Issue by Sarkie** [ValveSoftware/steam-for-linux#3301]
I am running elementary os 0.2 64 bit, when I press shut down either via the power button on the box or in the os, then Steam will shut down correctly as shown via running Steam via terminal.

What I have noticed is a timing difference, if Steam isn't running, this will be performed in under 4 seconds, if Steam is running it is greater than 25 seconds, even though according to the terminal output Steam has already taken the request and shut down Steam.

Is there any lingering processes that are running that are timing out after 20 seconds, are there any commands I can run to watch what is happening to Steam or elementary os to help see where the issue lies?

Processor Information:
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0x17
    CPU Stepping:  0x6
    CPU Type:  0x0
    Speed:  3000 Mhz
    2 logical processors
    2 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Unsupported

Network Information:
    Network Speed:  

Operating System Version:
    elementary OS Luna (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.8.0-39-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11203000
    X Window Manager:  Mutter(Gala)
    Steam Runtime Version:  steam-runtime-release_2014-04-15

Video Card:
    Driver:  X.Org Gallium 0.4 on AMD RV770

```
Driver Version:  3.0 Mesa 9.2.0
OpenGL Version: 3.0
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 74 Hz
VendorID:  0x1002
DeviceID:  0x9442
Number of Monitors:  2
Number of Logical Video Cards:  1
Primary Display Resolution:  1280 x 1024
Desktop Resolution: 2560 x 1024
Primary Display Size: 13.31" x 10.67"  (17.05" diag)
                                        33.8cm x 27.1cm  (43.3cm diag)
Primary VRAM Not Detected
```

Sound card:
    Audio device: Realtek ALC889A

Memory:
    RAM:  3952 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_GB.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  70935 Mb
    Largest Free Hard Disk Block:  19209 Mb

Installed software:

Recent Failure Reports:

**Comment by jjkinas** [comment]
This bug is also on Ubuntu Mate ! 

I must close steam before shutting down , or it is very slow  , same behavior.

**Comment by Zerophase** [comment]
Exists in Cinnamon too, on both Arch and Mint.

**Comment by calexil** [comment]
[Issue is ongoing](https://github.com/linuxmint/mate/issues/8)

**Comment by BenJetson** [comment]
I'm experiencing this issue as well on my new Ubuntu MATE 16.04 install. If the development team needs some log files or someone to test something, @ mention me and I'll be happy to help.

**Comment by calexil** [comment]
STILL ONGOING, Mint 18 mate

**Comment by Unaccounted4** [comment]
The exact same thing happens to me in Fedora 24 with Mate or Cinnamon, as well as Ubuntu Mate 16.04.1.

**Comment by stevethepocket** [comment]
Did the most recent client update fix this for all of you? They said they fixed something about shutting down with Steam open.

**Comment by Unaccounted4** [comment]
It didn't. It still delays the shutdown of MATE and Cinnamon by whatever amount SystemD is set to delay it for.

**Comment by SEAPUNK** [comment]
Yeah, which is strange. There must be some additional process that still doesn't listen to SIGINT, although Steam definitely appears to shut down on it. Maybe Steam takes too long to quit?

**Comment by calexil** [comment]
this issue was NOT resolved, mint 18.1 mate, still hangs for 30 seconds on shutdown if steam is open

**Comment by stevethepocket** [comment]
> Maybe Steam takes too long to quit?

This wouldn't surprise me at all. I've had issues in the past with Steam holding up the shutdown process in Windows.

**Comment by MartinX3** [comment]
+1

**Comment by Pandiora** [comment]
Same here, Issue still exists with Linux Mint 18.1 64-bit Cinnamon.

**Comment by MartinX3** [comment]
Same here, Issue still exists with Linux Mint 18.1 64-bit **Mate**.

**Comment by alextpedro** [comment]
Same issue. Linux Mint 18.2 Cinnamon 64-bit.

**Comment by Plagman** [comment]
Steam should obey SIGTERM now, any details on what exactly is happening? Is the Mint session relying on some other mechanism to instruct the client to exit?

**Comment by SEAPUNK** [comment]
My best guess is that Steam, on SIGTERM, takes too long to shut down, causing the system to wait (and eventually time out) on shutdown request. I don't 100% understand why the system doesn't catch Steam's process shutting down, unless there is another Steam process that keeps running in the background...

**Comment by Plagman** [comment]
Even with downloads running, the client always exits gracefully under a few seconds here if I manually send it SIGTERM.

**Comment by stevethepocket** [comment]
Something I am noticing now is that when I go to shut down while Steam is running, it forces Steam to quit, but then doesn't actually shut down; it just stays on a fully-functional desktop with everything except Steam still running. I have to pull up the shutdown option a second time. When I manually exit Steam first, and then hit the shutdown button, it shuts down properly the first time. This is on Kubuntu 16.04.

**Comment by kisak-valve** [comment]
I might as well add a note to that, with KDE on Arch, I see a notification from Plasma Workspace that says `Logout canceled by ''` when the steam window closes, but steam itself finishes closing a second or two later which is seen by the steam tray icon leaving the tray.

**Comment by Unaccounted4** [comment]
I have seen this very behaviour on KDE as well. On MATE it still does the same old thing, where shutdown becomes delayed by 90 seconds (which is the SystemD default timeout).

**Comment by frikisama** [comment]
Same behaviour here with Arch + XFCE. I think it only started recently, but can't pin it down to an exact date.

**Comment by gokudomatic** [comment]
Same behaviour with xubuntu and manjaro. It started to gobble the shutdown signal about 2 weeks ago.

**Comment by zatricky** [comment]
@frikisama Same scenario. Second week of July I came back to my PC a minute later to find it hadn't rebooted. Thought I just "didn't reboot". July 14th I realised it was a bug. For me I believe it started some time between July 5th and July 10th. I could be wrong of course. :-|

Going through pacman log I found very few xfce-related packages that had been updated recently. Even so, downgrading those packages did not help in any way.

**Comment by DrDoctor13** [comment]
Same issue with Manjaro and KDE. Plasma 5.10.3.

**Comment by mikerm19** [comment]
Same here with Linux Mint 18 (Cinnamon). It cancels the shutdown completely, but still exits Steam. I managed to screen shot the steam client shutting down window for a second after Cinnamon tried to shut down.

**Comment by MartinX3** [comment]
Switched to Ubuntu Mate 17.04.4
Same Problem, need to shutdown "twice".
(First one only exits steam.

**Comment by zatricky** [comment]
To add, happening on Fedora 25 as well, also with XFCE. I think it is safe to say this is not a distro-specific problem.

**Comment by wolfyrion** [comment]
I had also this problem with Steam and some other applications so I created a work around solution using Plasma Event Notification and Actions.

Run "Notifications"
Go to Event Source--> Plasma Workspace ===> Logout Canceled

In Plasma Notifications you can create a command to run when a "Logout  Canceled" event had happen.
So I just created a shut.sh script and made it executable  that includes 
#! /bin/bash
shutdown -P now

So every time a "Logout Canceled" event happens it just runs that script and it shuts down the pc...

**Comment by MartinX3** [comment]
@wolfyrion 
So every event which cancel the logout will shutdown the computer?
That's sound like you want to kill an insect with a gatling gun.

**Comment by zatricky** [comment]
@wolfyrion: I applaud your ingenuity - but @MartinX3 is right. The workaround also probably won't work unless you're using KDE.

The most likely bad scenario of course is where I click logout and an app shows that I hadn't saved something, prompting me to do the save. Except, too late, forced shutdown in progress. :-|

**Comment by wolfyrion** [comment]
@MartinX3 @zatricky : yeap you are right, only people with plasma can implement this work around. 
As for the bash script, I just posted what works for me but you can modify it according to your needs like for example to shutdown your PC after a min or secs, is a bash script so you can actually do whatever you want after a "Logout Canceled" event .
Well I am not using any other DE than KDE but even if I was using another DE I think I would create a shutdown button bash script , just to exit steam and shutdown the PC so as to make it easier for me until Steam fixes this problem.

(I use my pc for rendering or farming steam cards so I have to automatically shutdown my PC  because I am not always at home. Temperatures are a bit high these days  so I was using Kshutdown to schedule when the pc will be shutdown but unfortunately because of steam and some other apps my pc was not shutting down thats why I used that solution.)

**Comment by zatricky** [comment]
This issue appears to no longer affect XFCE+Arch for me - but does still appear to affect my XFCE + Fedora 25 install.

**Comment by Whitetigerswt** [comment]
Having this issue on Arch with KDE.

**Comment by Odel** [comment]
Having the same issue - Arch with KDE

Been ongoing for me since install about a year ago

**Comment by reconrey** [comment]
Same issue on Xubuntu 17.04, seemed to start around the time I upgraded from 16.10 to 17.04. Also, not sure if this is related, but around the same time my system started freezing on resume from suspend with Steam open.

**Comment by property-live** [comment]
is someone working on this? @triage-valve got assigned to this issue over 3 years ago and didn't even leave a comment yet.

**Comment by reconrey** [comment]
@property-live Yea, it is kind of ridiculous. However, considering us Linux uses make up less than one percent of the steam users ([http://store.steampowered.com/hwsurvey](http://store.steampowered.com/hwsurvey)) it only makes sense from a business perspective (though it kind of shows that even they do not have faith in their own Linux-based Steam Machines/SteamOS considering such a little bug has gone under the radar for so many years...).

**Comment by caligari87** [comment]
Commenting to confirm as well, still seeing this behavior on Linux Mint 18.2 Cinnamon. Pressing shutdown once closes Steam, but the system never subsequently shuts down unless I repeat the shutdown request with Steam not running. Closing Steam manually and then shutting down does not present any delays.

**Comment by SE2Dev** [comment]
I can confirm that this issue occurs on a relatively clean installation of Debian 9.1 (using XFCE) and I've experienced it before on Debian 8 using Cinnamon as well.

**Comment by efeciftci** [comment]
Confirmed on Kubuntu 16.04.3, I have to select shutdown twice if I have Steam client running. Latest Steam Beta is installed. This problem has started occuring since the last few weeks.

**Comment by ICHx** [comment]
I also confirm it on Ubuntu 17.04, OpenSUSE 42.3 and tumbleweed. No problem on Manjaro though.

**Comment by DrDoctor13** [comment]
Not sure if it means anything, but ending the session with something like Ctrl-Alt-Backspace or shutting down through a terminal doesn't cause this problem.

**Comment by BigMo** [comment]
Confirmed on Manjaro and KDE, Plasma 5.10.5. I extended @wolfyrion's approach and added both a notification and a 60s delay, allowing me to save any unsaved docs:

`#!/bin/bash

notify-send --app-name="Shutdown" "Shutdown failed; forced shutdown in 60 seconds"
sleep 60
shutdown -P now`

**Comment by SE2Dev** [comment]
I've begun attempting to reproduce this issue on virtual machines to see what actually causes it to occur.
Currently I've tried a fresh Debian 9.1 installation with Steam installed from the repo. The issue does not occur - in fact Steam automatically shuts down shortly after this dialog appears.
![image](https://user-images.githubusercontent.com/5873790/29826763-25af155e-8ca6-11e7-82e0-337975a41fbb.png)

I'm going to install some addition DE's and see if that has any affect.

**Comment by zatricky** [comment]
From what I've been able to find online where other projects have fixed similar issues, depending on what library Steam uses to interface with the DM, it may incorrectly return the close event with an "ignore", which causes the DM to cancel the logout. Alternatively, it might incorrectly return a "veto" to the close event, which is how applications normally prevent data loss with possibly-important unsaved content.

**Comment by SE2Dev** [comment]
@zatricky Yeah, that seems to be the case - I just installed xfce4 on the same system and logout / shutdown are canceled by Steam.

**Comment by zuh** [comment]
On Debian 9.1 with XFCE 4.12 it seems that the Steam logout dialog is interpreted as an error dialog and even though it cycles in and out on it's own (ie. completes logout/shutdown of Steam) it seems to return a value that xfce4-session sees as cancellation request. Which side of the fence the problem lays would require more digging, but maybe this helps someone to do it:

```

TRACE[xfsm-manager.c:1295] xfsm_manager_save_yourself_done(): enteringTRACE[sm-layer.c:232] sm_interact_request(): Client Id = 26d3bf123-032c-4cd7-97cd-aaa2114d9662, received INTERACT REQUEST [Dialog type = Error]
TRACE[sm-layer.c:246] sm_interact_done(): Client Id = 26d3bf123-032c-4cd7-97cd-aaa2114d9662, received INTERACT DONE [Cancel shutdown = True]
TRACE[xfsm-manager.c:303] xfsm_manager_set_state(): state is now XFSM_MANAGER_CHECKPOINT
TRACE[sm-layer.c:304] sm_save_yourself_done(): Client Id = 26d3bf123-032c-4cd7-97cd-aaa2114d9662, received SAVE YOURSELF DONE [Success = False]

```

**Comment by dtitov** [comment]
Same issue on the newest MacOS. It's been 3 years already since ticket creation: are the Steam-guys even looking at it?

**Comment by gvlfm78** [comment]
It doesn't look like it, it shouldn't even be too hard to fix. Maybe we can try contacting them directly if they pay no attention to the GitHub?

**Comment by bjwest** [comment]
This isn't really a three year old problem though, it started for me just a month or so ago.  I wish I could say if it were after a system update or Steam update, and which/when exactly but I don't really know.  Then only time I reboot my computer is when an update requires it.  Otherwise it's in hibernation when I'm not on it.

**Comment by IdleGandalf** [comment]
I think it was after a KDE update for me. Don't know which one it was through.

**Comment by bjwest** [comment]
I don't think KDE is the problem now.  I haven't updated my laptop in a while, it's using Plasma 5.9.4, KDE Framework 5.31.0 and QT 5.7.1 and rebooted with no problems.  Steam updated itself right after that, and now it's got this issue and won't reboot on the first try.  I wish I'd checked the Steam version before rebooting, but I only use my laptop for streaming games and only update Steam when it forces me to.  It updated to the one built July 17, API v017 and package version 1500335472.  A subsequent reboot updated Steam to an Aug 28 build and package version 1503954609, same API and same issue.

I'm pretty certain this is a Steam Issue now, most likely introduced in the July 17 build.

**Comment by Plagman** [comment]
The whole Steam process tree now closes promptly and gracefully when it gets a SIGTERM event. Steam has no awareness of higher-level inter-client communication around session shutdown, etc. If there's some such convention we're violating, we're not aware of it. Fixing the issue with SIGTERM allowed several major desktop environments to start logging out properly, so the remaining issues most likely are less general, and potentially related to behaviors specific to the desktop environments. If a developer for the affected desktop environments is able to give us information as to what behavior to change in order to avoid this issue, please attach it to this report or contact us directly, but just confirming the report or theorizing is not helpful.

Please don't bring up macOS as that is completely different and off-topic.

**Comment by SE2Dev** [comment]
@Plagman Are you referring to a previous update, the current version, or an upcoming version?

**Comment by Plagman** [comment]
Previous update; nothing has changed recently on that front.

**Comment by paulovilli** [comment]
I am running a fully updated Linux Mint Cinnamon 18.2 on a 4.11 Linux Kernel with AMD HD 7770 GPU and no closed source drivers.

When Steam is on the system tray and I ask the system to shutdown, Steam closes but shutdown sequence is aborted.

Its not just a long delay before shutdown carries on, Steam closes but somehow aborts the sequence. Then if I ask again for the system to shutdown it works immediately, as it does when Steam is manually "exited" before asking the system for a shutdown.

Despite old reports on Github the issue is quite recent for me (certainly 2017, probably since July).

**Comment by bfloch** [comment]
It seems this is where the message happens:
https://github.com/KDE/plasma-workspace/commit/616f57aecf6193efc333174360190549e2e4dcdb

As to what triggers it: vlc had similar issues in the past:
https://mailman.videolan.org/pipermail/vlc-commits/2011-July/007707.html

I am pretty sure the answer is hidden here:
https://www.x.org/releases/X11R7.6/doc/xorg-docs/specs/ICCCM/icccm.html
"Termination" is marked obsolete, but according to the vlc patch (from 2011) this was related to their issue.

**Comment by LinusCDE** [comment]
Just started recognizing this bug a few days ago for the first time.
I'm using Manjaro with the Nvidia-Drivers (v375.82).
It could have something to do with changing my Kernel from 4.8 to 4.12? Not completly sure on that but worth a try.

Just like *kisak-valve* [said above](https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-315560188):
> I might as well add a note to that, with KDE on Arch, I see a notification from Plasma Workspace that says Logout canceled by '' when the steam window visibly closes, but steam itself finishes closing a second or two later which is seen by the steam tray icon leaving the tray.

**Comment by renjithis** [comment]
I have been having this issue for a month or so now. I don't know where steam logs are, so please direct me to them if the following info is not enough.
Started steam from command line, waited for it to initialise. the following messages come after giving the shut down 
Ubuntu 16.04.3 with Cinnamon (no packages to be updated in software updater)
```
Refresh rate: 1
Refresh rate: 1
Refresh rate: 1
Refresh rate: 1
Refresh rate: 1
Refresh rate: 1
process 5993: arguments to dbus_connection_unref() were incorrect, assertion "connection->generation == _dbus_current_generation" failed in file ../../dbus/dbus-connection.c line 2794.
This is normally a bug in some application using the D-Bus library.
process 5993: arguments to dbus_connection_unref() were incorrect, assertion "connection->generation == _dbus_current_generation" failed in file ../../dbus/dbus-connection.c line 2794.
This is normally a bug in some application using the D-Bus library.
process 5993: arguments to dbus_connection_unref() were incorrect, assertion "connection->generation == _dbus_current_generation" failed in file ../../dbus/dbus-connection.c line 2794.
This is normally a bug in some application using the D-Bus library.
[2017-09-20 16:22:18] Startup - updater built Sep  6 2017 20:37:15
[2017-09-20 16:22:18] Verifying installation...
[2017-09-20 16:22:18] Verification complete
[2017-09-20 16:26:09] Shutdown
```
Same symptoms as others. On shutdown, steam exits, but shutdown is cancelled

**Comment by DrDoctor13** [comment]
New update dropped today for Linux. Can someone check to see if this somehow has changed anything? Changelogs:

> Linux
> Updated STEAM_RUNTIME_PREFER_HOST_LIBRARIES logic to still prefer Steam Runtime libraries if they are more recent than the host system
> Updated Vulkan loader in Steam Runtime to version 1.0.54, which provides all the extensions currently needed for SteamVR
> Added support for shader cache management on supported drivers (Mesa 17.1 and NVIDIA 381.26.08 or 384.59). Shader caches for games launched by Steam are placed in separate folders next to their Steam Library folder and are deleted when each game is uninstalled, or when switching drivers or graphics cards. Individual shaders are tracked and catalogued by the Steam servers in preparation for distributing pre-compiled shaders. This tracking can incur slight additional bandwidth use. The system can be disabled by setting the environment variable STEAM_ENABLE_SHADER_CACHE_MANAGEMENT=0
> Fix handling of XDG_DATA_DIRS and .desktop files with command-line arguments when adding non-steam games to your library
> Fixed adding arbitrary binaries as non-Steam shortcuts
> Fixed game support for DualShock 4 controllers when using recent kernels
> Fixed games using the ISteamHTMLSurface interface failing to display content on Linux if they created their browser object too quickly

**Comment by supagu** [comment]
having this issue on manjaro, kde edition.
Its a real pain. Any progress on a fix for it?

**Comment by reversesh3ll** [comment]
same. manjaro, kde edition.

**Comment by outfrost** [comment]
And same here. Shutdown attempt with Steam running - Steam exits, shutdown cancelled (no notification on this version of Plasma). Shutdown without Steam running proceeds as normal.

Here's some system info for reference:
![screenshot from 2017-10-07 22-29-59](https://user-images.githubusercontent.com/18421230/31311555-2d35686e-abaf-11e7-945d-e5070f1dd7b0.png)

**Comment by Photon89** [comment]
Arch Linux with Xfce here, same problem.

**Comment by Zamundaaa** [comment]
I have installed kubuntu-desktop after an upgrade to (beta) Ubuntu 17.10 . The error never occurred in Unity but now in KDE I have to do "logout" twice because apparently steam cancels it the first time

**Comment by zatricky** [comment]
I really wish it were easy to give more compact "me too"s in github issues. :-/

I've noticed the following behaviour and suspect strongly that it is the same issue. Steps:
- Ensure Steam is not running (can quit Steam or just do the next step before logging in to the computer)
- Disconnect network
- Launch steam
- Wait for Steam login to time out in order that it presents the option to "RETRY CONNECTION", "START IN OFFLINE MODE", or "QUIT".
- Though unnecessary for the test, you can now reconnect the network
- Attempt to log out or shutdown/reboot. Steam prevents this.
- tap "Quit" to quit Steam
- Attempt to log out or shutdown/reboot. Succeeds.

**Comment by SE2Dev** [comment]
@renjithis As I said before - the issue can be reproduced on a fresh installation of Debian 9 w/ Cinnamon and XFCE. I don't know why they've yet to address the issue when it can be reproduced on a clean system 100% of the time in a VM...

**Comment by renjithis** [comment]
Are we that much of a minority in an already minor portion of steam for linux users that they are ignoring us?
The last update didn't even have a Linux/SteamOS section, it only had genral, windows and mac. I thought we amounted more than Mac users who use Steam

**Comment by kisak-valve** [comment]
@renjithis, the current status of this issue is at https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-327368263. It's not clear what is triggering the bad behavior, even though it is reproducible.

**Comment by mikerm19** [comment]
So I'm not an expert, I'm willing to help identify. If anyone wants to point me to what log files are needed, etc I'm more than happy to collect information and post. I'm keeping my Linux Mint install very up to date along with the Steam client and the issue is still there. It's the only program out of everything I've tried that does it though.

**Comment by IdleGandalf** [comment]
Maybe we should bring this to the attention of the affected DE's developers by opening issues on the corresponding bug trackers. Don't think they gonna magically notice this here.

**Comment by renjithis** [comment]
I thought this was the place where issues with Steam for Linux is supposed to be reported (It still implies in the repo's description). Do we have a new place where we can report them? 

I was under the impression that confirmed issues will be posted by the repo admin/mods to the developer's bug tracker (which I would expect to be private).

**Comment by renjithis** [comment]
@kisak-valve  What more info can we provide that will help you (or who is working on this issue) identify the cause of the issue?

**Comment by IdleGandalf** [comment]
@renjithis I'm talking about the DE's developers, not steam. If you read the post way above by @Plagman he states that this will most likely not be fixed without the DE's help of tracking down what steam does wrong/ doesn't do.

**Comment by DrDoctor13** [comment]
If anyone makes a bug report at any DE's bugtracker, link it here so we can add ourselves to the CC lists/vote for the bug.

elementary OS/Pantheon: https://bugs.launchpad.net/elementaryos
MATE (not sure which repo would be appropriate): https://github.com/mate-desktop
Cinnamon: https://github.com/linuxmint/Cinnamon/issues
KDE: https://bugs.kde.org/
Xfce: https://bugzilla.xfce.org/
GNOME: https://bugzilla.gnome.org/

**Comment by zatricky** [comment]
https://bugzilla.xfce.org/show_bug.cgi?id=13841

Not sure if more information is needed there

**Comment by mikerm19** [comment]
Here's a thread recently started by someone else over in Mint land:

https://bugs.launchpad.net/linuxmint/+bug/1713601

**Comment by renjithis** [comment]
Has anyone opened issue with Cinnamon? Should I do it?

**Comment by property-live** [comment]
the latest release note say "Fixed an issue that could result in Steam blocking system shutdown or user logout."
has anyone tried it yet?

**Comment by kisak-valve** [comment]
Hello @property-live, that looks to be a Windows-only patch note (https://steamcommunity.com/groups/SteamClientBeta#announcements/detail/2971673654055775014) and did not change the client behavior on the test box I checked against.

**Comment by MartinX3** [comment]
I think, it helped on Ubuntu Mate x64 17.10.

**Comment by Scrumplex** [comment]
How can this issue persist for more than 3 years?

**Comment by mikerm19** [comment]
Unfortunately we are just in the super minority, so we are pretty low priority. I've been fastidious about keeping steam and mint completely up to date and it still cancels the download. Hopefully someday someone will fix for us.

**Comment by Managor** [comment]
Still an issue with xfce 4.12 on Ubuntu 17.10

**Comment by calexil** [comment]
[ongoing in mint 18.2 and 18.3 beta](https://www.reddit.com/r/linuxmint/comments/7hodld/does_hitting_quit_shutdown_only_close_steam_for/)

**Comment by paulovilli** [comment]
Retested after fresh install of Linux Mint 18.3 Cinnamon Edition (not beta)
and it is still happening here too.
Booting most recent versions of kernels 4.4, 4.8, 4.10, 4.11 and 4.13
available through the linux mint update manager made no diference. Those
are the Ubuntu Kernels pulled from kernel.ubuntu.org made available by
Canonical for Ubuntu 16.04 AFAIK.

It is important to note that right after asking for a power off (or reboot,
that also suffers from the same issue) I can very briefly see a dialog
window pop up on the center of the screen, saying it is syncing stuff and
asking me to confirm whether or not I want Steam to be force-closed
(something like that... too fast for me to read it properly or to take a
screenshot).

Then it closes, and the tray icon disappears, but the shutdown sequence is
halted.

Also, it seems that no othe software is ever closed before the sequence is
interrupted, so this all happens before linux tries to kill off anyone
remaining after asking nicely.


[Warning: all that follows bellow is observations and conjectures from an
attentive user but not a software programmer]

Finally I should add that I've seen something similar to this behavior on
Windows, where having notepad open to an unsaved but edited text file
prevents shutdown because the sequence is halted and aborted when a
save/close file dialog is offered by notepad to the user.

Maybe the fact that Steam is producing a new dialog window to interact with
the user is the trigger for the unwanted behavior. It does need to sync
saves to the cloud and stuff, but maybe it could wait for a while without
producing the dialog and only if this time limit is surpassed before
finishing sync it shows the dialog. This way a fast enough sync won't
produce a dialog window at all and maybe this prevents the issue...

**Comment by Managor** [comment]
As far as I know Ubuntu gnome 17.04 and Ubuntu mate aren't affected by this. They have a 30 second shutdown period where you can choose to shut to the computer down now or let it wait for 30 seconds.

**Comment by AdamSzopa** [comment]
> I might as well add a note to that, with KDE on Arch, I see a notification from Plasma Workspace that says Logout canceled by '' when the steam window visibly closes, but steam itself finishes closing a second or two later which is seen by the steam tray icon leaving the tray.

Yes! This is exactly what's happening on my system (kde neon) and it's driving me nuts! Please @kisak-valve, is there any way I can help provide more info? I'm seriously considering preventing steam from auto-starting, it is getting so infuriating.

**Comment by TecXero** [comment]
Still happening with xfce as well.

**Comment by Bednar87** [comment]
I am experiencing the same issuen as reported earlier in this threaed:

> [...]when I go to shut down while Steam is running, it forces Steam to quit, but then doesn't actually shut down; it just stays on a fully-functional desktop with everything except Steam still running. I have to pull up the shutdown option a second time. When I manually exit Steam first, and then hit the shutdown button, it shuts down properly the first time.

Arch XFCE.

**Comment by DevPre** [comment]
Very annoying behavior. Same result on Fedora 27 KDE.

**Comment by TheKoopaKingdom** [comment]
>EDIT: There is a KDE bug here: https://bugs.kde.org/show_bug.cgi?id=387587

But isn't this a bug in Steam? I've experienced this in XFCE, in addition to KDE.

**Comment by paulovilli** [comment]
Yes, despite the obvious signs that the bug is on Steam-for-Linux since it
affects multiple Distros and DEs, including Ubuntu (the only officially
supported distro), bug reports about this issue were opened for various
Distros and DEs.

If you read through the first few messages on this issue (and some other
similar issues on Valve's steam-for-linux Github), you will see that Valve
devs have on occasion manifested themselves. For the most part, they asked
those issues to be opened against each Distro/DE to get help from their
respective devs in identifying the cause for the issue.

Unfortunately I am not a Valve dev (I am not even a software dev at all) so
I don't know more the current status in their attempts to fix the issue.

In any case, if anyone keeping tabs on the issue against a Distro/DE gets
an answer from devs there, it would probably be good to alert here about
that.

2018-01-01 17:44 GMT-02:00 Koopa <notifications@github.com>:

**Comment by killerkalamari** [comment]
Occurs on Ubuntu MATE 16.04 LTS (64-bit).

**Comment by Zerophase** [comment]
Is one of us going to have to land a job at Valve to fix this? I'm going for a CS masters this fall, I think they recruit from Digipen, I could see about fixing the problem in two years.

**Comment by 3vi1** [comment]
Also still occurs on Ubuntu 18.04 (bionic beaver) development branch when using latest KDE plasma desktop packages from Ubuntu's devel repos.

**Comment by calexil** [comment]
@Zerophase lol

**Comment by outfrost** [comment]
> Is one of us going to have to land a job at Valve to fix this? I'm going for a CS masters this fall, I think they recruit from Digipen, I could see about fixing the problem in two years.

The funny thing is, had they open-sourced the client, they'd have 3 pull requests with this nonsense fixed like a year ago.

**Comment by efeciftci** [comment]
> That seems like it's pushing the problem on the DE devs, when Steam is the only app exhibiting this problem.

All of a sudden introduce a previously nonexistent new bug to your closed-source software and start blaming all the other open-source projects for this bug. Now that is something worth discussing about in Software Engineering courses in CS programs.

**Comment by Zerophase** [comment]
@outfrost I understand why they don't want to open source this would let competitors catch up, but then again might drop development costs. At least open source a small module for starting up and shutting down Steam.

**Comment by gvlfm78** [comment]
_what_ competitors?

**Comment by kisak-valve** [comment]
Instead of chaining second hand information, please refer back to https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-327368263. There is no attribution of blame, just a request for actionable help in resolving this issue instead of general confirmations that the issue exists.

**Comment by Photon89** [comment]
@kisak-valve: Could you (or somebody else) be more specific on which DEs allow proper shutdown? Reading through the comments, I have the impression that each single DE in existence (besides of maybe GNOME) is hit by this bug.

**Comment by Zerophase** [comment]
@gvlfm78 cd projekt red.

**Comment by stevethepocket** [comment]
I suspect @paulovilli is right about the culprit being that "Shutting down Steam..." window. KDE probably has some code that checks if any new windows are spawned in the split second after getting a shutdown command, in case a program wants to say "This file has not been saved. Save now?" And that's kind of what that window is for. If the sync and logout take more than a few seconds, it changes to a dialog asking if the user wants to keep trying or quit anyway.

So we're kind of at an impasse. You don't want Steam to just get rid of that window, because it might lose or even corrupt save data if syncing takes too long. And you don't want KDE to ignore it for the same reason. It could be tweaked so it _pauses_ the shutdown until any newly-spawned windows have been closed, and then resumes, but that would mess things up if users click "Don't Quit" on one of those unsaved-files dialogs.

The problem is, there's no standard for apps to signal directly that they want to delay a shutdown, and then later say to either resume or cancel it. And even if someone were to create one, any older apps that aren't compliant will be stuck losing unsaved data.

**Comment by MarcosCC** [comment]
@kisak-valve can't you reproduce the bug? I mean it clearly happens often enough, should be easy to reproduce and since you have the code, then follow it up to see what's going on.

In my case: install a fresh Debian + KDE, install Steam, open Steam, shutdown computer -> won't shut down. Only ever happens with Steam open.

**Comment by Scrumplex** [comment]
@stevethepocket there is actually a standard for delaying a shutdown and even a sleep. It is available in systemd 183 (available since 25 May 2012) and later and is called inhibit.
https://www.freedesktop.org/wiki/Software/systemd/inhibit/

**Comment by SEAPUNK** [comment]
Completely agreed w/ @MarcosCC. A lot of us Linux users are likely casual Linux users, and don't know all the various fancy internals of how the shutdown gets prevented. And considering how only Steam is the culprit here, this is likely an issue with how Steam handles shutdown. We don't have the ability to debug your Steam client, so we'd have no idea as to what Steam is doing wrong regarding its shutdown process.

**Comment by mikerm19** [comment]
I just want to put in a quick note, it's not just KDE that has this problem, at least Cinnamon, a fork of GNOME also has this problem, on both of my "daily driver" installs.

Everyone else who hasn't commented yet, I know it sucks to have to register an account to say "me too", but please do (it only takes a minute). We need as much feedback here as we can get. We are in a minority of Valve's problems here, so the bigger the voice, the better.

**Comment by SEAPUNK** [comment]
- [x] Cinnamon
- [x] KDE
- [x] XFCE
- [x] MATE
- [ ] GNOME

that's 4 out of 5 listed DEs doing it now

**Comment by Scrumplex** [comment]
@SEAPUNK you forgot Unity

**Comment by DrDoctor13** [comment]
Just to corroborate @SEAPUNK, I've tested all major DEs in a vanilla Ubuntu 17.10 virtual machine to see if the bug reproduces. To be clear, this is the procedure:

-Install Steam
-Log into Steam account
-Try to reboot computer from application menu
-Reboot is cancelled. Steam window pops up for a brief moment.
-Try to reboot a second time, reboot completes.

GNOME: Inconsistent. It seemed to stop the shutdown the first time but after repeated tries, Steam shutdown normally and the VM rebooted.
Xfce: Yes.
MATE: Yes.
KDE: Yes, with "Logout cancelled by ''" error.
Cinnamon (tested in Mint 18.3): Yes.
Unity: Unable to test (it hates me) but assuming similar behavior to GNOME.

**Comment by killerkalamari** [comment]
Is it possible to have Steam run in a mode where it launches a game and then exits after the game has finished? It already gives desktop launchers to run specific titles, and it seems to know how long I've been playing, so it must feel like it can tell when the game has been exited. This wouldn't be ideal for switching games because of the overhead, but it would be a good enough workaround for me. I would just have to remember that if I went to the steam store or wanted to chat that I'd need to close Steam manually afterwards.

**Comment by ffainelli** [comment]
At this point, it sounds like it might be more realistic to have a distribution/DE workaround that consists in sending a SIGTERM to steam (if running) prior to closing the DE session...

**Comment by outfrost** [comment]
Well what if Steam Cloud sync ran as a service? It wouldn't have to cancel nor inhibit desktop session shutdown in order to complete the sync, and init systems (such as systemd) have good support for processes that need a few extra seconds before terminating.

**Comment by scikirbypoke** [comment]
It seems my Debian sid 64-bit can shutdown properly after switched from xfce4 to LxQt 0.12

**Comment by VKNikov** [comment]
I can confirm experiencing this issue on Manjaro with KDE Plasman 5.11.5.

**Comment by codedevstem** [comment]
Confirmed bug on Fedora 27 with cinnamon spin.

**Comment by outfrost** [comment]
I believe that, at this point, Valve have more than enough information about distros and DEs to reproduce the issue.

**Comment by DrDoctor13** [comment]
Came across this systemd service to peacefully shut down Deluge. Could this be adapted for Steam to alleviate the problem?

https://gist.github.com/DrDoctor13/42168677f5c2a39594f5ab438c25cdb2

**Comment by paulovilli** [comment]
@ Buck Shockley

That script "peacefully" (read automatically and silently) *kills* deluge,
so it will not solve the issue exactly.

At best it is a workaround, but it might cause cloud savefile corruption,
desync or something like it because whatever Steam is doing when killed it
will stop in the middle of doing it.

We want Steam to be able to finish syncing savegames and whatnot to the
cloud before it closes.

We just don't want it to cause the system to give up on shutting down while
waiting for steam to finish that.

My suspicion is that when Steam opens a new dialog window while shutdown is
being processed it causes the system to stop the sequence.

So the best potential solutions yet were probably suggested by Sefa Eyeoglu
and Outfrost, when they remembered that systemd can receive some form of
"delay shutdown" signal from Steam.


On Jan 10, 2018 17:35, "Buck Shockley" <notifications@github.com> wrote:

Came across this systemd service to peacefully shut down Deluge. Could this
be adapted for Steam to alleviate the problem?

https://gist.github.com/DrDoctor13/42168677f5c2a39594f5ab438c25cdb2

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-356712203>,
or mute the thread

**Comment by masieden** [comment]
Void, xfce4 checking in.

**Comment by DrDoctor13** [comment]
@paulovilli What if SIGTERM is sent to Steam instead of SIGKILL? Running `pkill -15 steam` seems fine.

**Comment by paulovilli** [comment]
(re)disclaimer: I am not a software programer, just a nosey end-user

that being said, isn't that the same signal sent by the system to all
processes when you ask it to shutdown?

I believe it would then have the same outcome if the signal is sent by a
dedicated systemd service (like the one you mentioned) or by the
shutting-down system (afaik handled by systemd too?)...

but maybe not, this is waaaay over my level of knowledge!


On Jan 16, 2018 13:13, "Buck Shockley" <notifications@github.com> wrote:

@paulovilli <https://github.com/paulovilli> What if SIGTERM is sent to
Steam instead of SIGKILL? Running pkill -15 steam seems fine.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-357991000>,
or mute the thread

**Comment by DrDoctor13** [comment]
@paulovilli Not quite. SIGKILL will kill the process immediately and not give it a chance to clean up. SIGTERM also kills the process, but gives it a chance to clean up, i.e. finish syncing or pause downloads.

**Comment by outfrost** [comment]
My proposal is to have a separate, background service just for Steam Cloud Sync (and any other tasks that need to be stopped gracefully) that would run completely separate from any user logins and DE sessions. The desktop Steam client could simply connect to that service to carry out its sync duties (exchanging authentication info in the process) and use it for that purpose whenever needed. In fact, many clients could connect to that same service, each from a different system and/or Steam user, and be distinguished between by the service. This would not only ensure that the desktop client doesn't need to spawn a window and cancel shutdown (and that it can thus be killed safely), but it would also have the added benefit of Steam Cloud Sync and anything else included in the service being shielded from desktop client crashes, sudden display server restarts, etc. On shutdown, the DE could then simply exit immediately, while the init system would wait for the Steam service to get its stuff wrapped up.

**Comment by SE2Dev** [comment]
I just threw together a small program that logs the signals it receives just before it exits; It received SIGHUP when I tried to log off (in Cinnamon). Is the issue perhaps that non-Gnome environments are sending different signals (e.g. SIGTERM)?

**Comment by paulovilli** [comment]
Is it a bash script? can you share the code here somehow so we can test
which signal each distro/DE is sending?

Just to confirm... SIGHUP is the signal sent to Steam by Cinnamon (Linux
Mint? which version? which kernel version?) as soon as you ask the system
to shutdown?


From this explanation (
http://meinit.nl/the-3-most-important-kill-signals-on-the-linux-unix-command-line)
I would guess usual shutdown behavior would be sending SIGHUP at first then
some more aggressive signals after some elapsed time.

The weird thing is, despite the description from
https://en.wikipedia.org/wiki/Signal_(IPC), Steam didn't react to Ctrl+D
when ran from terminal (Ctrl+D sent with terminal window in focus), not
even returning any message in the terminal.

When I tried Ctrl+C to send it SIGINT it also did not close, just returned
this message and kept going: "^CForced create but already created for
SharedObjectEvent". That means it is programed to handle SIGINT internally
somehow, right?

When I tried Ctrl+\ to send it SIGQUIT it closed immediately.


2018-01-18 23:26 GMT-02:00 SE2Dev <notifications@github.com>:

**Comment by SE2Dev** [comment]
@paulovilli 
```
Debian GNU/Linux 9.3 (stretch)
Linux version 4.9.0-5-amd64
Cinnamon 3.2.7
```

I just meant I had a simple C program that printed when it received a signal - just so I could figure out what the DE was sending the processes. The code is literally:
```c++
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>

FILE *logfile = NULL;

#define LOG(...)         \
    printf(__VA_ARGS__); \
    fprintf(logfile, __VA_ARGS__)

void startup(void)
{
    logfile = fopen("log.txt", "w");
}

int shutdown(int exit_code = EXIT_SUCCESS)
{
    LOG("[NOTE]: exiting...\n");
    fflush(logfile);
    fclose(logfile);
    return exit_code;
}

void signal_callback_handler(int signum)
{
    LOG("[SIGNAL]: %d\n", signum);
    exit(shutdown(signum));
}

int main()
{
    startup();

    // Let's just register the callback handler for all possible signals
    for (int sig = 1; sig < _NSIG; sig++)
    {
        signal(sig, signal_callback_handler);
    }

    for (int i = 0;; i++)
    {
        // Print a line every so often so we know the program was actually running
        LOG("[IDLE]: %d\n", i);
        sleep(1);
    }

    return shutdown();
}
```

I'm curious what this will print for Gnome - since the issue doesn't seem to occur with Gnome.

**Comment by h1z1** [comment]
The cloudsync isn't the only thing to "block" Steam.  Crashuploads for example also cause this (uploads to crash.steampowered.com).  It's a creepy little background service running with steam that calls home whenever something breaks (files are in /tmp/dumps for the curious).  Since it's yet another host valve doesn't enable ssl on, interception is trivial.    Point is Steam will block while it tries. 

The other thing to consider is not all signals are delivered in realtime, which means they can block both in an interactive shell and background.  So you can end up sending SIGINT all you want for example, it won't 
matter if the process is sleeping or blocked on something else. 

There is no magic bullet, shutdowns are hard.  Anyone whose had to deal with NFS knows this :)   I'd rather window managers tell me what process is still blocked then blindly kill things.  We have runlevels and different signals for a reason.

At any rate there are threads like [this](https://bbs.archlinux.org/viewtopic.php?id=179489) pointing fingers at more general problem not unique to Steam.

**Comment by paulovilli** [comment]
Is it possible to find out if Steam is doing a permanent block (inhibit
block) instead of a temporary one (inhibit delay) in reply to being asked
to exit (by means of SIGHUP/SIGINT/SIGTERM or whatever X11 window closing
comands it actually receives)?

Do those issued signals and corresponding blocks get logged somewhere? Is
there a way to monitor it?


Maybe STEAM is handling its signal by using inhibit(block) instead of
inhibit(delay)?
https://www.freedesktop.org/wiki/Software/systemd/inhibit/

If that is the case then either:

1- it is using inhibit(delay) at first and then changing to inhibit(block)
when that dialog window comes up expecting user interaction...

...this could mean if it just waits 2 seconds longer before that window
comes up most users would stop having issues.

or
2- it is just treating itself like an essential piece of software and using
inhibit(block) right from the start...

...this might explain why I noticed no other software ever gets closed
before the shutdown sequence is aborted?

Again, all conjecture from my part, I am no expert at all :(

**Comment by Leeo97one** [comment]
The worst part of all this is that absolutely nothing has been proposed by the Steam developers since the opening of the problem.

**Comment by megatog615** [comment]
>The worst part of all this is that absolutely nothing has been proposed by the Steam developers since the opening of the issue.

I've noticed that Valve sometimes ignores UI bugs(this is somewhat a UI bug, I guess) when a major version is planned to come out soon that completely overhauls the UI. No need to fix the bug if it isn't even present in the overhaul. See: Dota 2 when the new in-game HUD came out.
Valve has hinted that it's redesigning the look of Steam's interface.

**Comment by 3vi1** [comment]
> I've noticed that Valve sometimes ignores UI bugs(this is somewhat a UI bug, I guess) when a major version is planned to come out soon that completely overhauls the UI. 

"Meh... we're going to redo that in 4 or 5 years... ignore their rather large problem until then.  What are they going to do?  Install Desura?!?  lol"

**Comment by Plagman** [comment]
The new client Beta should fix this, can people affected re-test?
http://steamcommunity.com/groups/SteamClientBeta#announcements/detail/1655505073049286411

**Comment by John-Gee** [comment]
It fixed it for me using Plasma!

Thanks!

**Comment by sh4nks** [comment]
The latest beta update fixed the issue for me!

```
Operating System Version:
    "Arch Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.14.15-1-ARCH
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11906000
    X Window Manager:  KWin
    Steam Runtime Version:  <Runtime disabled>
```

**Comment by frikisama** [comment]
Latest Beta fixes the issue for me too. Arch + XFCE here, using native runtime.

**Comment by zuh** [comment]
Beta client on XFCE (Debian) and shutdown works normally again.

Thank you very much for the developers for finding a fix for this!

**Comment by Zerophase** [comment]
Shutdown still takes about 30 seconds on Cinnamon. Cinnamon notices an "unknown" program is trying to shutdown, steam closes in the tray, and then the desktop icons disappear. About 30 seconds later the desktop exits.

**Comment by Managor** [comment]
Fix works on Xubuntu.

**Comment by Leeo97one** [comment]
OK this looks good. But what's the point of auto-closing Steam when the shutdown menu appears under GNOME?
See: https://github.com/ValveSoftware/steam-for-linux/issues/3301#issuecomment-325685239

**Comment by stevethepocket** [comment]
Fixed for me on Kubuntu 17.10. Before this gets closed and forgotten forever, would you mind disclosing what the fix was? I'm sure other developers could benefit from the insight.

**Comment by bjwest** [comment]
@Zerophase  That's a normal function of any modern desktop environment, there's a lot going on that needs to be cleanly shutdown.  We haven't been able to just turn off the computer since the Win 3.x days.

**Comment by Zerophase** [comment]
@bjwest I definitely get that. It still takes a ton longer to shut the desktop down then if I had manually exited steam before shutting down. I could understand syncing slowing the process down, but that isn't running since I don't have any games installed.

On windows steam shuts down fairly fast. 

I just think it's worth looking in on Cinnamon to see if there are any other issues delaying shutdown.

**Comment by paulovilli** [comment]
I have a Phenom II x4 965 running Linux Mint Cinnamon 18.3 with kernel
4.13.0-32, all freshly updated, with a AMD HD7770 GPU with no proprietary
drivers installed (actually the most recent AMD CPU microcode yes, but none
for the GPU):
https://gist.github.com/anonymous/1feaa7063988e021d2b3b8c196f10846

I have not received the update with the fix yet. I'm not using steam beta.
I asked Steam to check for an update but it says there is none. Was it
supposed to have arrived yet?
Steam client version is built 15/dec/2017 11:47:57, API v017 and package
v1513371133.


@Zerophase
I timed my computer shutting down right now so I can have a baseline to
compare when the fix comes in... First shutdown attempt takes 2sec to close
Steam (tray icon disappearing) and second attempt issued 1sec later takes
6sec to power off the PC.

**Comment by kisak-valve** [comment]
Hello @paulovilli, until the next stable client bump happens, you'll need to opt into the Steam beta client if you want the fix for this issue.

`Steam` -> `Settings` -> `Change...` button in Beta participation -> `Steam Beta Update` in the dropdown

**Comment by paulovilli** [comment]
Thanks @kisak-valve for the clarification!

I'll post back the results about shutdown timing in Cinnamon.

If it increases notably should me and @Zerophase keep talking about it here
or open a new issue and reference this one?

**Comment by Zerophase** [comment]
@paulovilli Hmm, I'm going to have to time my actual shutdown time, but it takes me significantly longer. (when not closing steam manually) 

I'm using steam-native-runtime, and have the latest beta installed.

**Comment by Dinoraptor101** [comment]
Same issue here
Manjaro Linux
KDE Plasma 5.11.5
KDE Frameworks 5.42.0
QA Vesion: 5.10.0
Kerlen: 4.14.14-1-MANJARO
64 bit archetecture
CPU: Intel I5-2500k.

Shutdown is completely cancelled. (logout canceled by plasma)

**Comment by Photon89** [comment]
@Dinoraptor101: Problem is fixed in Steam beta (at least for me), either wait till the fix reaches stable or switch to beta in the Steam settings.

**Comment by outfrost** [comment]
Steam beta update seems to have resolved or worked around the issue for me. Shutdown takes about 3-5 seconds longer to leave the desktop session, but it doesn't get cancelled.
`Mint 18.2 / KDE Plasma 5.8.8 / Linux 4.8.0-53-generic amd64`

**Comment by paulovilli** [comment]
@Zerophase and @kisak-valve

I have tested Steam Beta today and the fix works (Steam no longer aborts
shutdown sequence) but there is indeed something wrong with it taking so
long to shutdown now...

I timed it to 38 seconds now, of which only 5 are before Steam's tray icon
disappears and almost all of it is staring at the desktop with barely no
signs that the shutdown sequence is still running.

Just as a reminder, I have posted a week ago about how I timed my computer
shutting down to have a baseline to compare when the fix came in... First
shutdown attempt took 2sec to close Steam (tray icon disappearing) and
second attempt issued 1sec later took 6sec to power off the PC. So even
counting the time lost with a secondary human interaction it took a lot
less time to shutdown before the beta fix.

I will try to post back tomorrow after testing other kernel versions to see
if there is any relation, though it seems unlikely.

Just to make it clear, the fix works. It just needs polishing to improve
the speed of shutdown which got a bit too long on LM Cinnamon 18.3
(apparently only there). Should we open a new issue and reference this one
or just keep posting here about it?


Distro: Linux Mint Cinnamon 18.3
Kernel: Ubuntu Kernel 4.13.0-32 (LM 18 is based on Ubuntu 16.04 repos so
this is the latest HWE kernel available from kernel.ubuntu.org)
CPU: Phenom-II x4 (with latest AMD CPU microcode installed - version
2.20160316.1)
GPU: AMD HD7770 (with no proprietary drivers)
No PPAs installed
Steam: beta - jan/25/2018 21:25:07
Steam API: v018
Steam package: 1516948201
https://gist.github.com/anonymous/1feaa7063988e021d2b3b8c196f10846


2018-02-04 19:06 GMT-02:00 outfrost <notifications@github.com>:

**Comment by kisak-valve** [comment]
@paulovilli, yes, please open a new issue report for your follow up issue.

**Comment by athordan** [comment]
Thank you Valve!
Signing in for Beta worked fine for me.
Manjaro - KDE Plasma 5

**Comment by paulovilli** [comment]
For anyone experiencing much longer (+20/30 seconds) shutdown times after
the fix for this issue, I've just opened a new follow-up issue here:
https://github.com/ValveSoftware/steam-for-linux/issues/5379#issue-295363828

(so far only Linux Mint Cinnamon is reportedly affected)

2018-02-06 9:52 GMT-02:00 athordan <notifications@github.com>:

**Comment by mikerm19** [comment]
Can confirm, beta now lets Cinnamon (Linux Mint) shut down, thanks!

It does however make Cinnamon pop up a box saying that it's waiting on an unknown program to shut down, so there is other issues, maybe for #5379.

**Comment by mlewis-everley** [comment]
I can also confirm that the beta version allows shutdown for me (Linux Mint + Cinnamon), but I also seem to suffering from issue #5379  (same as @paulovilli and @mikerm19)

**Comment by Canule** [comment]
Can confirm, beta now lets Cinnamon (Linux Mint) shut down, thanks!

Like mikerm19 said:

It does however make Cinnamon pop up a box saying that it's waiting on an unknown program to shut down, so there is other issues, maybe for #5379.

**Comment by EarthDevour** [comment]
It is finally fixed ! Kubuntu 17.10 latest steam client !

**Comment by kisak-valve** [comment]
Closing per "Fixed an issue where the Steam client could prevent some desktop sessions from shutting down" in the [2018-03-21](http://store.steampowered.com/news/38412/) steam client update.

**Comment by paulovilli** [comment]
hey there @Martin!

What you describe is a different issue from what this bug was about.

I did a quick search for issues involving system standby and found none was
similar either:
https://github.com/ValveSoftware/steam-for-linux/search?q=standby&type=Issues

This aborted powerdown issue did pop up on the search, then 2 issues about
NOT preventing standby while on big picture mode/using xbox controller.

I suggest you open a new issue.

Just be sure to give the issue an accurate and detailed description, and
give all requested info (a suggested issue report layout will appear when
you create it).

PS: I am not a Valve developer, just a user trying to help.
